### PR TITLE
fix(store): stop a store named specs or changes becoming the root

### LINKED ISSUE #1881: A store named `specs` or `changes` at `~/openspec/<id>` makes `$HOME` the planning root
## Description

The stores guide places a store at `~/openspec/<id>`. When the id is `specs` or
`changes`, the store's folder *is* `~/openspec/specs` or `~/openspec/changes`.

Root selection classifies a directory as a real planning root when its
`openspec/specs` or `openspec/changes` exists as a directory. So `$HOME` now
qualifies. From anywhere under the home directory:

- `$HOME` wins as the `nearest` root, and the global `defaultStore` is never
  consulted
- `new change` writes into `~/openspec/changes/`, outside every store. With id
  `changes`, that is loose inside the store's own top-level folder

`src/core/root-selection.ts:277-283` states that this exact situation is what the
qualifying walk exists to prevent:

> Without this, the recommended `~/openspec/<id>` store layout would make $HOME
> a phantom root that captures every command under the home tree.

`specs` is an ordinary name for a planning store.

## Reproduction

```bash
openspec store setup specs --path ~/openspec/specs --json
openspec config set defaultStore specs
mkdir -p ~/src/web-app && cd ~/src/web-app
openspec list --json
openspec new change probe-change --json
```

## Expected

`root` is `{"source": "global_default", "store_id": "specs"}`, and the change is
created at `~/openspec/specs/openspec/changes/probe-change`, as it is for any
other store id.

## Actual

```
[team-plans] list root: source=global_default path=.../openspec/team-plans store=team-plans
             new change -> .../openspec/team-plans/openspec/changes/probe-change
[specs]      list root: source=nearest path=<$HOME> store=None
             new change -> <$HOME>/openspec/changes/probe-change
[changes]    list root: source=nearest path=<$HOME> store=None
             new change -> <$HOME>/openspec/changes/probe-change
```

## Root cause

`classifyOpenSpecDir` (`src/core/project-config.ts:590-594`) counts any existing
`openspec/specs` or `openspec/changes` directory as planning shape. It does not
notice that the directory is itself a store root (it carries
`.openspec-store/store.yaml`).

## Impact

For a user whose store happens to be called `specs` or `changes`, every command
anywhere under the home directory acts on `$HOME` instead of the configured
store, and new changes are written outside any store. Nothing warns.

## Environment

- OpenSpec 1.13.0 (reproduced on the published npm package and on a build of
  `main` @ `9d4e597`)
- Node 22, Linux
- Every run used its own temporary `HOME`, `XDG_CONFIG_HOME` and `XDG_DATA_HOME`

### PR DESCRIPTION
**Pull request by dwin-gharibi**
Closes #1881.

## Why

The stores guide places a store at `~/openspec/<id>`. With the id `specs` or
`changes`, the store's own folder *is* `~/openspec/specs` or
`~/openspec/changes`.

`classifyOpenSpecDir` treated any existing `openspec/specs` or
`openspec/changes` directory as a planning shape, so `$HOME` qualified as a real
root. From anywhere under the home directory:

- `list`, `status`, `new change` and the rest resolved `$HOME` as the `nearest`
  root, and the global `defaultStore` was never consulted
- `new change` wrote into `~/openspec/changes/`, outside every store

`src/core/root-selection.ts` states that the qualifying walk exists to prevent
exactly this: "the recommended `~/openspec/<id>` store layout would make $HOME a
phantom root that captures every command under the home tree". The walk
handled a bare `~/openspec/` but not these two ids.

## What Changes

- `classifyOpenSpecDir` no longer counts a `specs/` or `changes/` directory as
  planning shape when that directory carries store metadata
  (`.openspec-store/store.yaml`). Such a directory is a store root, not planning
  content of the directory above it.
- The change is inside the one shared classifier, so root resolution, the
  `init` pointer guard, `store setup`'s pointer check and `doctor` keep agreeing,
  as that function's comment requires.

Unchanged:

- a real project's `openspec/specs/` and `openspec/changes/`
- resolution from inside the store itself
- a `$HOME` whose `openspec/changes/` is real planning stays a root, even when
  `openspec/specs/` beside it is a store

### Why detect by metadata rather than by the registry?

A store folder is a store whether or not this machine has registered it yet, for
example a fresh clone. The metadata check also keeps `classifyOpenSpecDir`
synchronous and free of registry reads, like the rest of the root walk.

## Testing

`test/commands/store-phantom-root.test.ts`: 15 tests. 9 run through the built
CLI; 6 are unit tests of `classifyOpenSpecDir`. Written first and watched fail:

- **Before the fix (tests only, on `9d4e597`):** 7 failed, 8 passed.
  - For the `specs` and `changes` stores, `root` came back as
    `{ path: <$HOME>, source: 'nearest' }` instead of the store's
    `global_default`.
  - `new change` did not land in the store.
  - `classifyOpenSpecDir` returned `true` for a store-root folder.
  - Every control passed.
- **After the fix:** 15 passed.

### CI parity

Run in an isolated sandbox on `9d4e597` under Node 20.19.0 and pnpm 10.34.5,
with the steps from `.github/workflows/ci.yml`:

| Step | Result |
|---|---|
| `pnpm run build` | pass |
| `pnpm exec tsc --noEmit` | pass |
| `pnpm lint` | pass |
| `pnpm test` (`VITEST_MAX_WORKERS=4`) | 4565 passed, 12 failed (4577) |
| `pnpm test` on clean `9d4e597`, same sandbox | 4555 passed, 8 failed (4563) |

The full-suite failures are all timeouts — `Test timed out in 10000ms`, or
`spawnSync npm ETIMEDOUT` in the npm packing test — in CLI-heavy files this
change does not touch: `store-references`, `store-remote`,
`store-root-selection`, `workset`, `package-install-scripts` and
`store.test.ts`. Two checks say the sandbox causes them, not this change:

- clean `9d4e597` fails the same files the same way on the same machine
- every one of those files passes alone (`--maxWorkers=2`), on this branch and
  on clean `9d4e597`: `store-references` 8/8, `store-remote` 19/19,
  `store-root-selection` 44/44, `store.test.ts` 43/43, `workset` 41/41,
  `package-install-scripts` 8/8

At that run the branch added exactly its 14 new tests: 4577 = 4563 + 14. The symlinked-alias test added since brings the file to 15.

Edge cases covered:

- control: a store named `team-plans` resolves as `global_default`
- stores named `specs` and `changes` resolve as `global_default`, with the right
  path and `store_id`
- `new change` lands in the store for both ids, and nothing is written under
  `$HOME/openspec/changes/`
- from inside the store, the store itself is the `nearest` root
- a real project at `$HOME` still wins as `nearest`, ahead of `defaultStore`
- `$HOME` with a real `openspec/changes/` beside a `specs` store stays a root
- from a symlinked alias of the home tree, the `specs` store still resolves as `global_default` at its canonical path (skipped on Windows, where directory symlinks need elevation)
- unit tests:
  - a plain `openspec/specs` or `openspec/changes` counts as planning
  - a store-root `openspec/specs` or `openspec/changes` does not
  - when only one of them is a store root, the other still counts
  - store metadata deeper inside `openspec/specs` does not disqualify it

## Changeset

`.changeset/store-named-specs-is-not-a-root.md` (patch).

## Summary by CodeRabbit

- **Bug Fixes**
  - Fixed project root detection for stores named `specs` or `changes`.
  - Commands run from home-directory workspaces now resolve the configured default store correctly.
  - New changes are saved inside the intended store instead of an unintended home-directory location.
  - Existing project planning directories and genuine project roots continue to resolve as expected.

- **Tests**
  - Added coverage for store resolution, change creation, and planning-directory classification in these scenarios.

### CODE REVIEW

**Comment by clay-good** [comment]
Hardening pass, one commit on top of the fix:

- 0e45c7c0: canonicalized the root identity assertions in `store-phantom-root.test.ts` and added a symlinked-alias case (CodeRabbit thread).

Verified: the issue repro on main resolves `$HOME` as `nearest` and writes `~/openspec/changes/probe-change`; on this branch it resolves `global_default` / `specs` and the change lands in the store. The test file fails 7/15 with `src/core/project-config.ts` reverted and passes 15/15 with it. Root-selection, project-config, init and doctor suites pass (289/289); tsc and lint clean. The new `project-config` -> `store/foundation` import adds no cycle.

File: test/commands/store-phantom-root.test.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    expect(setup.exitCode).toBe(0);
+    const config = await runCLI(['config', 'set', 'defaultStore', id], { cwd: tempDir, env });
+    expect(config.exitCode).toBe(0);
+    return storeRoot;
+  }
+
+  async function rootFrom(cwd: string): Promise<any> {
+    const result = await runCLI(['list', '--json'], { cwd, env });
+    expect(result.exitCode).toBe(0);
+    return parseJson(result).root;
```
[the remark this answers is not in the document]
Comment (clay-good) [reply test/commands/store-phantom-root.test.ts]: Fixed in 0e45c7c0: `rootFrom` canonicalizes the returned `root.path` with `fs.realpathSync.native()`, every expected `storeRoot`/`home` is canonicalized too, and a new case (skipped on Windows, where directory symlinks need elevation) runs from a symlinked alias of the home tree and asserts the same canonical `global_default` store root.

**Review by alfred-openspec — APPROVED** [review]
The shared root classifier now distinguishes store metadata from project planning directories without registry coupling, including symlinked paths and mixed real-planning controls. Focused tests pass.
